Add support for view, sequence, column & function privileges

biodevc opened the pull request:
Currently only TABLE privileges (aka roles, permissions) are handled.

This adds support for VIEWS, SEQUENCES, COLUMNS & FUNCTIONS.

biodevc committed:
Include VIEWS in privileges (aka role, permission)

Currently only table privileges are handled

biodevc committed:
Update privileges.sql

biodevc committed:
Also include SEQUENCE & COLUMN privileges

biodevc committed:
Add support for function priveleges

biodevc commented:
I just saw that  @michaelbeaumont had done the groundwork for function privs so copied and tweaked his query.
It should address #djrobstep/migra#64

provokateurin commented:
Could you also add schema privileges? They are missing too.

provokateurin commented:
I implemented the schema privileges. @biodevc could you pickup https://github.com/stack11/schemainspect/commit/2107f082613def968615dc0e48413b9555193056 in your PR?

biodevc committed:
Add support for schema priveleges

As per @jld3103

biodevc committed:
Add support for schema privileges

As per @jld3103

biodevc commented:
> I implemented the schema privileges. @biodevc could you pickup [stack11@2107f08](https://github.com/stack11/schemainspect/commit/2107f082613def968615dc0e48413b9555193056) in your PR?

Just added your mods but altered the query since it was producing wierd results for me

provokateurin commented:
Huh what results? We've been using this patch in production and it always worked for us.

biodevc committed:
Merge branch 'master' into patch-1

biodevc committed:
Update privileges.sql

biodevc commented:
> Huh what results? We've been using this patch in production and it always worked for us.

If a role has 'group' members and that role has priveleges granted to a schema, then with your query, all members of the role also get granted priveleges to the schema.  So, it "works", but if a member is later removed from the top level role, they'll still have access to the schema.

biodevc committed:
Update inspected.py

provokateurin commented:
@biodevc could you rebase?

joshainglis commented:
A small change that is worth adding to this PR for function privileges to work (needs to include the function signature)
https://github.com/joshainglis/schemainspect/commit/14a1d7a6a00dee203033da31461e619b31f2e3b3

biodevc commented:
> @biodevc could you rebase?

I've given up hope of this pull request ever going anywhere. Migra appears to be a dead project at this stage.

provokateurin commented:
yeah :( but i thought i saw a post somewhere about the maintainer working on these projects again.
